src/shm_ext.c

```diff
@@ -131,26 +131,27 @@ sr_shmext_conn_remap_unlock(sr_conn_ctx_t *conn, sr_lock_mode_t mode, int ext_lo
 {
     sr_error_info_t *err_info = NULL;
     sr_ext_hole_t *iter, *last = NULL;
-    uint32_t last_size;
-    size_t shm_file_size = 0;
+    size_t shm_file_size;
+    void *last_hole_end;
 
     /* make ext SHM smaller if there is a memory hole at its end */
     if (((mode == SR_LOCK_WRITE) || (mode == SR_LOCK_WRITE_URGE)) && ext_lock) {
         while ((iter = sr_ext_hole_next(last, SR_CONN_EXT_SHM(conn)))) {
             last = iter;
         }
 
-        if (last && ((uint32_t)((char *)last - conn->ext_shm.addr) + last->size == conn->ext_shm.size)) {
+        last_hole_end = last ? last + last->size : NULL;
+
+        if (last_hole_end == conn->ext_shm.addr + conn->ext_shm.size) {
             if ((err_info = sr_file_get_size(conn->ext_shm.fd, &shm_file_size))) {
```

Comment:
@michalvasko For some reason `conn->ext_shm.size` and `sr_file_get_size(conn->ext_shm.fd, &shm_file_size)` get out of sync, causing the failure in test_process, although we have a write lock

Reply:
Interesting that it manifests only in the external DB tests. I have no ideas so you will have to investigate this yourself, or I can later.

Reply:
I am checking it.

Reply:
I believe this was the reason:

```c
last_hole_end = last ? last + last->size : NULL;
```
is really not the same as 
```c
last_hole_end = last ? (char *)last + last->size : NULL;
```
as `*last` is `sr_ext_hole_t` size and not a `char` size.

Reply:
Sorry, I meant
```C
last_hole_end = last ? ((char *)last + last->size) : NULL;
```
to have the whole operand of the ternary operator in parentheses.

Reply:
Ah, okay. Done